### ISSUE TITLE
exclude all COFF binaries from "symbol_presence" test

### ISSUE THREAD

**Issue by petrovr** [openssl/openssl#17194]
Some release 3.0 tests fail on CYGWIN:
- symbol_presence
- abort

Request is for first one.

Remark: In corrections for issue #13558 looks like that MINGW is excluded by accident (mistake). For good this is right decision.

**Comment by petrovr** [comment]
For protocol details for failed tests:
```
....
01-test_symbol_presence.t .......... 1/?
# Number of lines in @nm_lines before massaging: 0
# Number of lines in @def_lines before massaging: 5377
# Number of lines in @nm_lines after massaging: 0
# Number of lines in @def_lines after massaging: 5373
# The following symbols are missing in cygcrypto-3.dll:
#   ACCESS_DESCRIPTION_free
#   ....
#   v2i_GENERAL_NAME_ex
not ok 2 - check that there are no missing symbols in cygcrypto-3.dll
01-test_symbol_presence.t .......... 3/? ---------------------------------------
# Number of lines in @nm_lines before massaging: 0
# Number of lines in @def_lines before massaging: 521
# Number of lines in @nm_lines after massaging: 0
# Number of lines in @def_lines after massaging: 517
# The following symbols are missing in cygssl-3.dll:
#   BIO_f_ssl
# ....
#   i2d_SSL_SESSION
not ok 4 - check that there are no missing symbols in cygssl-3.dll
# ------------------------------------------------------------------------------
#   Failed test 'check that there are no missing symbols in cygssl-3.dll'
#   at test/recipes/01-test_symbol_presence.t line 135.
01-test_symbol_presence.t .......... Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/4 subtests
....
```
```
...
01-test_abort.t ....................
test/aborttest.c:14: OpenSSL internal error: Voluntary abort
*** starting debugger for pid 20495, tid 2088
../../util/wrap.pl ../../test/aborttest.exe => 0
not ok 1 - Testing that abort is caught correctly
01-test_abort.t .................... 1/? ---------------------------------------
#   Failed test 'Testing that abort is caught correctly'
#   at test/recipes/01-test_abort.t line 16.
#          got: '1'
#     expected: '0'
01-test_abort.t .................... Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests
....
```

**Comment by petrovr** [comment]
The request to exclude "symbol test" is because test look like is oriented to ELF targets and nm utility from binutils.

NM is not suitable to this tests when is used COFF binary. Also test is skipped in 1.1.1 because it search for lib[foo].so  while used is cyg[foo].dll or lib[foo].dll.

----
Some resutls from nm command:
$ nm cygcrypto-3.dll  | grep ' [BDT] ' | grep ACCESS_DESCRIPTION_free
0000.... T ACCESS_DESCRIPTION_free

$ nm cygcrypto-3.dll  | grep ' [BDT] ' | grep v2i_GENERAL_NAME_ex
000.... T v2i_GENERAL_NAME_ex

$ LANG=C nm -DPg cygcrypto-3.dll
nm: cygcrypto-3.dll: **no symbols**

$ LANG=C nm -Pg cygcrypto-3.dll  | grep ACCESS_DESCRIPTION_free
ACCESS_DESCRIPTION_free T ...

$ nm cygcrypto-3.dll  | grep ' [BDT] ' | grep ACCESS_DESCRIPTION_free
00000005e03ce020 T ACCESS_DESCRIPTION_free

$ nm cygcrypto-3.dll  | grep ' [BDT] ' | grep v2i_GENERAL_NAME_ex
00000005e03d5230 T v2i_GENERAL_NAME_ex

$ LANG=C nm -DPg cygcrypto-3.dll
nm: cygcrypto-3.dll: no symbols

$ nm -Pg cygcrypto-3.dll |grep ' [BCSDT] '  | wc -l
7308

$ nm cygssl-3.dll | grep ASN1_OCTET_STRING_free
000.... I __imp_ASN1_OCTET_STRING_free
000.... T ASN1_OCTET_STRING_free

As is visible from above commands NM cannot be used for COFF binary. 

----
For protocol results from 1.1.1 branch
```
../test/recipes/01-test_symbol_presence.t ..
1..4
# NOTE: developer test!  It's possible that it won't run on your
# platform, and that's perfectly fine.  This is mainly for developers
# on Unix to check that our shared libraries are consistent with the
# ordinals (util/*.num in the source tree), something that should be
# good enough a check for the other platforms as well.
ok 1 # skip Can't run 'nm -DPg ../../libcrypto.so' => 256...  ignoring
ok 2 # skip Can't run 'nm -DPg ../../libcrypto.so' => 256...  ignoring
ok 3 # skip Can't run 'nm -DPg ../../libssl.so' => 256...  ignoring
ok 4 # skip Can't run 'nm -DPg ../../libssl.so' => 256...  ignoring
ok
All tests successful.
```

**Comment by t8m** [comment]
This is fixed with #17119

**Comment by petrovr** [comment]
Closed?
